pipert/core/shared_memory.py

```diff
@@ -90,43 +104,51 @@ def get_shared_memory_object(name):
 
 class SharedMemoryGenerator:
     """
-    Generates a new shared memory each time get_next_shared_memory is called
-    and is responsible for cleaning up shared memories if the count that
-    exists now exceeds the max or the proccess has ended.
+    Generates a set 'max_count' amount of shared memories to be used.
     """
-    def __init__(self, component_name, max_count=50):
-        self.memory_id_gen = MemoryIdGenerator(component_name, max_count)
+    def __init__(self, component_name, max_count=50, size=5000000):
+        self.memory_id_gen = MemoryIdIterator(component_name, max_count)
         self.max_count = max_count
         self.shared_memories = {}
+        for _ in range(self.max_count):
+            next_name = self.memory_id_gen.get_next()
+            memory = posix_ipc.SharedMemory(next_name, posix_ipc.O_CREAT,
+                                            size=size)
+            semaphore = posix_ipc.Semaphore(next_name, posix_ipc.O_CREAT)
+            mapfile = mmap.mmap(memory.fd, memory.size)
+            memory.close_fd()
+
+            semaphore.release()
+            self.shared_memories[next_name] = SharedMemory(memory, semaphore,
+                                                           mapfile)
+
+    def get_next_shared_memory(self):
+        """
+        Iterates over the existing memories in a cycle.
 
-    def get_next_shared_memory(self, size=5000000):
-        next_name, name_to_unlink = self.memory_id_gen.get_next()
-
-        memory = posix_ipc.SharedMemory(next_name, posix_ipc.O_CREAT,
-                                        size=size)
-        semaphore = posix_ipc.Semaphore(next_name, posix_ipc.O_CREAT)
-        mapfile = mmap.mmap(memory.fd, memory.size)
-
-        memory.close_fd()
-
-        self.shared_memories[next_name] = SharedMemory(memory, semaphore,
-                                                       mapfile)
-
-        if name_to_unlink:
-            if name_to_unlink in self.shared_memories:
-                self._destroy_memory(name_to_unlink)
-
-        semaphore.release()
+        Returns: The next shared memory that's available to use.
 
-        return next_name
+        """
+        return self.memory_id_gen.get_next()
 
     def cleanup(self):
+        """
+        Cleans all of the allocated shared memories to free up the ram.
+
+        """
         for _ in range(self.max_count):
```

Comment:
I think that if you used iterator pattern it could've been much more readable and easier to do (without even knowing `self.max_count`).

Let's do it together.